### ISSUE TITLE
The currently selected adapter `@astrojs/cloudflare` is not compatible with the image service "Sharp".

### ISSUE THREAD

**Issue by hkbertoson** [withastro/adapters#191]
### Astro Info

```block
Astro                    v4.4.13
Node                     v20.11.0
System                   macOS (arm64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             none
```

### Describe the Bug

The currently selected adapter `@astrojs/cloudflare` is not compatible with the image service "Sharp".

### What's the expected result?

I have set my configuration to use 'compile'. Which according to the docs. Uses Astro’s default service (sharp), but only on pre-rendered routes at build time. During SSR for pages rendered on-demand, all astro:assets features are disabled.

### Link to Minimal Reproducible Example

https://github.com/hkbertoson/sleepy-spiral

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by alexanderniebuhr** [comment]
It's reproduced and should be fixed. But it is not breaking anything, it's just a warning.

**Comment by danawoodman** [comment]
I'm also getting this error with Astro + CloudFlare adapter and the passthrough image service, assuming this error shouldnt even be possible if im not using sharp?

```ts
import cloudflare from "@astrojs/cloudflare";
import tailwind from "@astrojs/tailwind";
import { defineConfig, passthroughImageService } from "astro/config";
import sitemap from "@astrojs/sitemap";

export default defineConfig({
  output: "hybrid",
  integrations: [tailwind(), sitemap()],
  adapter: cloudflare(),
  image: { service: passthroughImageService() },
});
```

**Comment by alexanderniebuhr** [comment]
Yeah it's a warning, which we would like to hide, but it's a core upstream change, so it might take some time. Still we appreciate all the reports and feedback

**Comment by florian-lefebvre** [comment]
This should be fixed in 5.0, see https://5-0-0-beta.docs.astro.build/en/guides/upgrade-to/v5/#changed-properties-on-supportedastrofeatures-adapter-api

**Comment by nlnw** [comment]
It is not fixed in 5.0. It's a problem with the cloudflare adapter.

**Comment by florian-lefebvre** [comment]
I think that's because the adapter is not using the new shape yet, @alexanderniebuhr may confirm

**Comment by alexanderniebuhr** [comment]
Yes we still need to update some things

**Comment by millette** [comment]
With Astro 5.0.3 and @astrojs/node v9.0.0, there's a similar warning for the node adapter:

> [WARN] The currently selected adapter `@astrojs/node` is not compatible with the image service "Sharp".

**Comment by florian-lefebvre** [comment]
Can you update a dedicated issue for it? That would help thank you!

**Comment by nlnw** [comment]
This is not fixed even with the latest release.

**Comment by Uninen** [comment]
Using Astro 5.1.6 and @astrojs/cloudflare 12.2.0 I now get not one but _two_ warnings of this when running the dev server:

```
22:36:16 [WARN] [config] The adapter @astrojs/cloudflare has limited support for "sharp". Certain features may not work as expected.
22:36:16 [WARN] [adapter] Cloudflare does not support sharp. You can use the `compile` image service to compile images at build time. It will not work for any on-demand rendered images.
```

(the project has no `sharp` anywhere)

**Comment by alexanderniebuhr** [comment]
@Uninen the warning you get is expected.

**Comment by ekwoka** [comment]
@alexanderniebuhr  Why is it expected if the settings are telling it not to use sharp?

**Comment by alexanderniebuhr** [comment]
The 'compile' option uses sharp, if switching to a different option which doesn't use sharp, the warning should go away.

**Comment by ekwoka** [comment]
But with it set to compile..
It then says it should be set to compile.

So the warning says to use compile when you already are using compile.

It also shows when using cloudflare passthrough

**Comment by alexanderniebuhr** [comment]
@ekwoka as I said for 'compile' it is correct to have a warning, maybe we can tweak the wording a bit. For 'passtrough' it is not expected, I need to look into that case.

**Comment by yocheco** [comment]
Using Astro 5.2.3 and @astrojs/cloudflare 12.2.1 

`
12:06:09 [WARN] [config] The adapter @astrojs/cloudflare has limited support for "sharp". Certain features may not work as expected.
12:06:09 [WARN] [adapter] Cloudflare does not support sharp. You can use the `compile` image service to compile images at build time. It will not work for any on-demand rendered images.
`
